CMSgov/design-system
# [WNMGDS-1273] Convert Healthcare Footer to TypeScript

## Summary
https://jira.cms.gov/browse/WNMGDS-1273

### Changed
- Convert HC.gov Footer and all subcomponents to TypeScript

### Fixed
- Update the i18next-related code to be compatible with its new API
  - Unfortunately this means perhaps misusing its namespaces feature in order for us to be able to keep our own API unchanged. I have made a note about it in the code.

## How to test
View the storybook story for the footer to test it. Make sure switching between English and Spanish works.
